services/core-api/app/api/securities/resources/bond.py

```diff
@@ -0,0 +1,88 @@
+from flask_restplus import Resource, reqparse
+from datetime import datetime
+from flask import current_app, request
+from werkzeug.exceptions import BadRequest, NotFound, InternalServerError
+from marshmallow.exceptions import MarshmallowError
+
+from app.extensions import api, db
+from app.api.securities.response_models import BOND
+from app.api.securities.models.bond import Bond
+from app.api.utils.access_decorators import requires_role_view_all, requires_role_edit_bonds
+from app.api.utils.resources_mixins import UserMixin
+from app.api.mines.permits.permit.models.permit import Permit
+from app.api.mines.mine.models.mine import Mine
+
+
+class BondListResource(Resource, UserMixin):
+    @requires_role_view_all
+    @api.marshal_with(BOND, envelope='records', code=200)
+    def get(self):
+
+        mine_guid = request.args.get('mine_guid', None)
+
+        if mine_guid is None:
+            raise BadRequest('Please provide a mine_guid.')
+
+        mine = Mine.find_by_mine_guid(mine_guid)
+
+        if mine is None:
+            raise BadRequest('No mine found with the provided mine_guid.')
```

Comment:
raise NotFound instead

Reply:
Changed

Reply:
I actually think this should return empty set. given it's a querystring parameter.